<launch>

  <arg name="mav_name" default="bluebird" />
  <arg name="namespace" default="$(arg mav_name)" />

  <!-- The namespace all node occupy-->
  <group ns="$(arg namespace)" >

    <!-- Playing back the data -->
    <node name="rosbag_play" type="play" pkg="rosbag" output="screen"
      args=" -q /home/millanea/trunk/vicon_odometry/vrpn_client_fault_finding/data/2016/velocity_check.bag">
      <remap from="vrpn_client/estimated_transform" to="disused1" />  
      <remap from="vrpn_client/estimated_odometry" to="disused2" />  
      <remap from="vrpn_client/vicon_intermediate_results" to="disused3" />  
    </node>

    <!-- Ros Vrpn Estimation node -->
    <node name="vicon_estimator" type="vicon_estimation_node" pkg="ros_vrpn_client" output="screen">
      <param name="object_name" value="$(arg mav_name)" />
      <param name="vicon_estimator/dt" value="0.01" />
      <param name="translational_estimator/kp" value="1.0" />
      <param name="translational_estimator/kv" value="0.1" />
      <param name="rotational_estimator/orientation_estimate_initial_covariance" value="1" />
      <param name="rotational_estimator/rate_estimate_initial_covariance" value="1" />
      <param name="rotational_estimator/orientation_process_covariance" value="0.00001" />
      <param name="rotational_estimator/rate_process_covariance" value="10" />
      <param name="rotational_estimator/orientation_measurementCovariance" value="0.005" />
      <param name="rotational_estimator/outlier_threshold_degrees" value="30" />
      <param name="rotational_estimator/maximum_outlier_count" value="50" />
      <param name="rotational_estimator/output_minimal_quaternions" value="false" />
    </node>

    <node name="rosbag_record" type="record" pkg="rosbag" output="screen"
      args=" -o /home/millanea/trunk/vicon_odometry/vrpn_client_fault_finding/output/ /$(arg mav_name)/vicon_estimator/estimated_odometry /$(arg mav_name)//vicon_estimator/estimated_transform /$(arg mav_name)/vicon_estimator/vicon_intermediate_results /$(arg mav_name)/vrpn_client/raw_transform">
    </node>

  </group>

</launch>